### PR TITLE
Hotfox/fix ci

### PR DESCRIPTION
**Pull request by yanganto**
- fix ci flow
  - the ci will run and fold the log the tests of Darwinia, so the Travis log can be easy to use
- fix some test case in Kton pallet

### CODE REVIEW

**Comment by yanganto** [comment]
There are still some test cases not ready in Kton pallet.  The PR here is to prevent the conflict on CI flow.

**Comment by yanganto** [comment]
Here is the current result in Kton pallet.
```
running 6 tests
test tests::without_transfer_fee::remove_lock_should_work ... ok
test tests::without_transfer_fee::combination_locking_should_work ... ok
test tests::without_transfer_fee::transfer_should_work ... ok
test tests::without_transfer_fee::set_lock_should_work ... ok
test tests::without_transfer_fee::transfer_should_fail ... ok
test tests::without_transfer_fee::update_lock_should_work ... ok

test result: ok. 6 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```
The behavior of `set_lock_should_work` is under checking, and `remove_lock_should_work` is not total ready.

**Comment by HackFisher** [comment]
```
error[E0599]: no function or associated item named `set_lock` found for struct `Module<mock::Test>` in the current scope
   --> frame/balances/ring/src/tests.rs:44:14
    |
44  |               Balances::set_lock(ID_1, &1, 9, u64::max_value(), WithdrawReasons::all());
    |                         ^^^^^^^^ function or associated item not found in `Module<mock::Test>`
    | 
   ::: frame/balances/ring/src/lib.rs:423:1
    |
423 | / decl_module! {
424 | |     pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
425 | |         type Error = Error<T, I>;
426 | |
...   |
552 | |     }
553 | | }
    | |_-[0m function or associated item `set_lock` not found for this
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use darwinia_support::traits::LockableCurrency;`
```

**Comment by AurevoirXavier** [comment]
> ```
> error[E0599]: no function or associated item named `set_lock` found for struct `Module<mock::Test>` in the current scope
>    --> frame/balances/ring/src/tests.rs:44:14
>     |
> 44  |               Balances::set_lock(ID_1, &1, 9, u64::max_value(), WithdrawReasons::all());
>     |                         ^^^^^^^^ function or associated item not found in `Module<mock::Test>`
>     | 
>    ::: frame/balances/ring/src/lib.rs:423:1
>     |
> 423 | / decl_module! {
> 424 | |     pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
> 425 | |         type Error = Error<T, I>;
> 426 | |
> ...   |
> 552 | |     }
> 553 | | }
>     | |_-[0m function or associated item `set_lock` not found for this
>     |
>     = help: items from traits can only be used if the trait is in scope
>     = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
>             `use darwinia_support::traits::LockableCurrency;`
> ```

```rust
use darwinia_support::traits::LockableCurrency;
```

**Comment by AurevoirXavier** [comment]
> ```
> error: failed to run custom build command for `node-runtime v0.4.0 (/home/travis/build/darwinia-network/darwinia/bin/node/runtime)`
> Caused by:
>   process didn't exit successfully: `/home/travis/build/darwinia-network/darwinia/target/debug/build/node-runtime-3f3fdb6692974ff6/build-script-build` (exit code: 1)
> --- stdout
> Executing build command: "cargo" "rustc" "--target=wasm32-unknown-unknown" "--manifest-path=/home/travis/build/darwinia-network/darwinia/target/debug/wbuild/node-runtime/Cargo.toml" "--color=always" "--release"
> --- stderr
>    Compiling wasm-build-runner-impl v1.0.0 (/home/travis/build/darwinia-network/darwinia/target/debug/wbuild-runner/node-runtime)
>     Finished dev [unoptimized + debuginfo] target(s) in 1.37s
>      Running `/home/travis/build/darwinia-network/darwinia/target/debug/wbuild-runner/node-runtime/target/debug/wasm-build-runner-impl`
>    Compiling clear_on_drop v0.2.3
>    Compiling backtrace-sys v0.1.32
>    Compiling arrayvec v0.4.12
> error: failed to run custom build command for `clear_on_drop v0.2.3`
> Caused by:
>   process didn't exit successfully: `/home/travis/build/darwinia-network/darwinia/target/debug/wbuild/target/release/build/clear_on_drop-9185fd336fa16725/build-script-build` (exit code: 1)
> --- stdout
> TARGET = Some("wasm32-unknown-unknown")
> OPT_LEVEL = Some("3")
> HOST = Some("x86_64-unknown-linux-gnu")
> CC_wasm32-unknown-unknown = None
> CC_wasm32_unknown_unknown = None
> TARGET_CC = None
> CC = None
> CFLAGS_wasm32-unknown-unknown = None
> CFLAGS_wasm32_unknown_unknown = None
> TARGET_CFLAGS = None
> CFLAGS = None
> CRATE_CC_NO_DEFAULTS = None
> DEBUG = Some("false")
> running: "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=wasm32-unknown-unknown" "-Wall" "-Wextra" "-o" "/home/travis/build/darwinia-network/darwinia/target/debug/wbuild/target/wasm32-unknown-unknown/release/build/clear_on_drop-3b80494220e46872/out/src/hide.o" "-c" "src/hide.c"
> cargo:warning=error: unable to create target: 'No available targets are compatible with this triple.'
> cargo:warning=1 error generated.
> exit code: 1
> --- stderr
> ```

fixed in https://github.com/darwinia-network/darwinia/pull/259/commits/b75f44123feadd3f12aeefc37d6e8bc6ffd98b14

**Comment by yanganto** [comment]
All test cases in Kton is ready
```
running 6 tests
test tests::without_transfer_fee::combination_locking_should_work ... ok
test tests::without_transfer_fee::lock_release_then_remove_lock_then_lock_again_should_work ... ok
test tests::without_transfer_fee::set_lock_should_work ... ok
test tests::without_transfer_fee::transfer_should_work ... ok
test tests::without_transfer_fee::transfer_should_fail ... ok
test tests::without_transfer_fee::update_lock_should_work ... ok
```

File: frame/balances/ring/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ frame-support = { version = "2.0.0", default-features = false, git = "https://gi
 frame-system = { version = "2.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", rev = "c2fccb36ffacd118fc3502aa93453580a07dc402" }
 sp-runtime = { version = "2.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", rev = "c2fccb36ffacd118fc3502aa93453580a07dc402" }
 sp-std = { version = "2.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", rev = "c2fccb36ffacd118fc3502aa93453580a07dc402" }
+pallet-balances = { version = "2.0.0", git = "https://github.com/paritytech/substrate.git", rev = "c2fccb36ffacd118fc3502aa93453580a07dc402" }
```
Comment (AurevoirXavier) [inline frame/balances/ring/Cargo.toml]: unused

Comment (yanganto) [reply frame/balances/ring/Cargo.toml]: Thanks.  :)

**Comment by HackFisher** [comment]
two of the Build jobs still failing.

https://travis-ci.org/darwinia-network/darwinia/jobs/649375257?utm_medium=notification&utm_source=github_status

**Comment by AurevoirXavier** [comment]
```rust
error: couldn't read /home/travis/build/darwinia-network/darwinia/target/release/build/node-runtime-03ddd373f9b0a69f/out/wasm_binary.rs: No such file or directory (os error 2)
  --> bin/node/runtime/src/lib.rs:75:1
   |
75 | include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

This's a problem from cargo cache. Make some changes of the code and the `wasm_binary.rs` will be rebuild. I'll find a way to fix this.

**Comment by yanganto** [comment]
Native and wasm build pass

**Comment by yanganto** [comment]
The CI fails because the job exceeded the maximum time limit for jobs, and has been terminated.

File: .travis.yml [hunk 2]
```diff
@@ -13,17 +13,40 @@ branches:
...
-        - RUST_TOOLCHAIN=nightly TARGET=native
 
 before_install:
     # Check how much space we've got on this machine.
     - df -h
 
-script:
-    - .maintain/ci/script.sh
+jobs:
+  include:
```
Comment (HackFisher) [inline .travis.yml]: It takes quite long time, does it do env clean before each stage? I mean, does every stage require rebuilding?

**Comment by yanganto** [comment]
Ready to merge.
https://travis-ci.org/darwinia-network/darwinia/builds/649972453?utm_medium=notification&utm_source=github_status

File: .travis.yml (hunk 2 above)
Comment (yanganto) [reply .travis.yml]: I cache `target` to solve this issue.

**Pull request merged**